apps/recorder/app.js

```diff
@@ -311,7 +311,7 @@ function plotTrack(info) { "ram"
   g.drawString(/*LANG*/"Back",g.getWidth() - 10, isBTN3 ? (g.getHeight() - 40) : (g.getHeight()/2));
   setWatch(function() {
     viewTrack(info.fn, info);
-  }, isBTN3?BTN3:BTN1);
+  }, isBTN3?BTN3:BTN1, {edge:"falling"});
```

Comment:
No :P

From the software reference: 
`{ repeat: true/false(default), edge:'rising'/'falling'/'both'(default), debounce:10}`

But I'm not entirely sure that was the problem - since I think the default of `repeat:false` would mean it should only run once. But I suspect, didn't confirm it, that graph/plot screen would catch the `"rising"` and the menu one level up would catch the `"falling"` edge. Thus causing the behavior I observed.

Reply:
But I agree. It sounds likely that there were a time when the code didn't result in the double-back behaviour.

Reply:
I suspect watching the rising edge creates a problem because `E.showMenu` calls `Bangle.setUI` which sets up a watch for the falling edge. So the one button press triggers the rising watch for `viewTrack` and on the falling edge then the `back` handler of `setUI` because the menu and its falling edge watch is setup before the button is releases. That can be tested by calling `viewTrack` in a timeout. As long as the button is released before `viewTrack` is called it works fine.

Your solution of using the falling edge should work fine for now, the button press (rising and falling) is completely done before setting up the menu -> Maybe additional documentation for this behaviour is needed somewhere?
It seems weird that using a `setWatch` with defaults before `setUI` creates unexpected behaviour by always triggering whatever `setUI` set up for the same button. Maybe `setUI` should use the rising edge as well?

Reply:
tagging @gfwilliams so he can chime in on this.

Reply:
I think reacting on the falling edge is nice because you're not leaving a rising edge for the next screen/app/whatever to potentially misfire on. It makes the UI a little snappier by reacting on the rising edge though.

Reply:
Yes, the leftover falling edge when reacting to rising in `setUI` could be problematic. Actually the same way it is here... 😅
Changing the default for buttons to falling could be possible, but that has the risk of creating problems for all apps actually needing the rising edge AND using the defaults. No easy way to tell how many that would be.

Reply:
> Are you sure? I guess you verified it? but seems counter to what I think I've learned while working on different things. I'll change to edge rising instead of falling and test it on my watch.

I think I was not clear enough, with your fix rising and falling is done and can not be picked up by the watch from `setUI`.

Reply:
Yeah sorry - I could have read it better 😛

Reply:
Yes, responding on the falling edge looks good - and was certainly the reason for going back 2 levels. 

I forget the original reasoning for using falling in Bangle.js (I think someone wanted to be able to detect difference between long/short presses?) but now everything uses falling, so really most apps should try and use that where possible to avoid this issue.

I'd prefer not to change the `setWatch` default just because it's so low level and shared with other devices, but I think ideally we should just try and use `setUI` wherever we can and avoid this and the whole `isBTN3?BTN3:BTN1` issue (I should really update the tutorials)

Reply:
Yes, using `setUI` wherever possible is definitely the way to go.